lib/carrierwave/sequel.rb

```diff
@@ -11,6 +11,43 @@ def mount_uploader(column, uploader, options={}, &block)
       raise "You need to use Sequel 3.0 or higher. Please upgrade." unless ::Sequel::Model.respond_to?(:plugin)
       super
 
+      mod = Module.new
+      include mod
+      mod.class_eval <<-RUBY, __FILE__, __LINE__+1
+        def #{column}=(new_file)
+          if !(new_file.blank? && send(:#{column}).blank?)
+            modified!
+          end
+          super
+        end
+
+        def remove_#{column}!
```

Comment:
AR version calls `write_#{column}_identifier` in `remove_#{column}!`. Sequel version calls `write_#{column}_identifier` in `before_save` https://github.com/carrierwaveuploader/carrierwave-sequel/blob/master/lib/carrierwave/sequel.rb#L69
If `modified!` is not called, `before_save` will not be executed unless `save` is explicitly called (i.e. won't work with `update`).
If you have the time it would be of course better to recreate the sequel plugin to match the activerecord one more closely.